models/DataObject/ClassDefinition/Data/ReverseManyToManyObjectRelation.php

```diff
@@ -296,6 +273,190 @@ public function isOptimizedAdminLoading(): bool
+                throw new \InvalidArgumentException('Object is not an instance of an allowed class in field "'.$this->getName().'"');
+            }
+
+            $reverseObjects = $item->get($ownerFieldName);
+            foreach((array)$reverseObjects as $reverseObject) {
+                if($reverseObject->getId() === $object->getId()) {
+                    continue 2;
+                }
+            }
+            $reverseObjects[] = $object;
+
+            $item->set($ownerFieldName, $reverseObjects);
+        }
+
+        $oldRelationIds = array_map(function(DataObject\Concrete $newRelation) {
+            return $newRelation->getId();
+        }, $oldRelations);
+        $newRelationIds = array_map(function(DataObject\Concrete $newRelation) {
+            return $newRelation->getId();
+        }, (array)$data);
+        $deletedRelationIds = array_diff($oldRelationIds, $newRelationIds);
+
+        foreach($deletedRelationIds as $deletedRelationId) {
+            $deletedRelation = DataObject\Concrete::getById($deletedRelationId);
+            $reverseObjects = $deletedRelation->get($ownerFieldName);
+
+            foreach($reverseObjects as $index => $reverseObject) {
+                if($reverseObject->getId() === $object->getId()) {
+                    unset($reverseObjects[$index]);
+                }
+            }
+            $deletedRelation->set($ownerFieldName, array_values($reverseObjects));
+        }
+
+        return parent::preSetData($object, $data, $params);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function prepareDataForPersistence($data, $object = null, $params = [])
+    {
+        $db = Db::get();
+
+        $oldRelations = $db->fetchAll('SELECT src_id FROM object_relations_' . $this->getOwnerClassId().' WHERE dest_id=? AND fieldname=? AND ownertype = \'object\'', [$object->getId(), $this->getOwnerFieldName()]);
+        $oldRelations = array_map(function($oldRelation) {
+            return $oldRelation['src_id'];
+        }, $oldRelations);
+
+        $deletedRelationIds = $oldRelations;
+        $newRelationIds = array_map(function(DataObject\Concrete $newRelation) {
+            return $newRelation->getId();
+        }, (array)$data);
+        $deletedRelationIds = array_diff($deletedRelationIds, $newRelationIds);
+
+        foreach($deletedRelationIds as $deletedRelationId) {
+            $deletedRelation = DataObject\Concrete::getById($deletedRelationId);
+
+            $reverseObjects = $deletedRelation->get($this->getOwnerFieldName());
+
+            foreach($reverseObjects as $index => $reverseObject) {
+                if($reverseObject->getId() === $object->getId()) {
+                    unset($reverseObjects[$index]);
+                }
+            }
+            $deletedRelation->set($this->getOwnerFieldName(), array_values($reverseObjects));
+
+            $version = $deletedRelation->saveVersion(true, true, $params['versionNote'] ?? null);
```

Comment:
Saving a version even versioning is disabled ?

Reply:
Oh, certainly there is a check missing. Which disabling do you mean? I only know of the number of version items in system config.

Reply:
exactly.

Reply:
@weisswurstkanone what do you mean with "versioning is disabled"? In https://github.com/pimcore/pimcore/blob/b87de319676a7978f0f1e73d0163133d86eda114/models/DataObject/Concrete.php#L285-L289 there is the actual check for the number of versions to be saved. Or is there any other place where you can disable versioning except setting the number of versions to be saved?

Reply:
AFAIK, there is no central helper for that kind of decision. Please still keep in mind that only the published version of the data should go into the database.